iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergInputFormat.java

```diff
@@ -177,9 +178,19 @@ public boolean shouldSkipCombine(Path path, Configuration conf) {
 
   @Override
   public VectorizedSupport.Support[] getSupportedFeatures() {
-    // disabling VectorizedSupport.Support.DECIMAL_64 as Parquet doesn't support it, and we have no way of telling
-    // beforehand what kind of file format we're going to hit later
-    return new VectorizedSupport.Support[]{  };
+    throw new UnsupportedOperationException("This overload of getSupportedFeatures should never be called");
+  }
+
+  @Override
+  public VectorizedSupport.Support[] getSupportedFeatures(HiveConf hiveConf, Properties properties) {
+    // disabling VectorizedSupport.Support.DECIMAL_64 for Parquet as it doesn't support it
+    boolean isORCOnly = Boolean.parseBoolean(properties.getProperty(HiveIcebergMetaHook.ORC_FILES_ONLY));
```

Comment:
We also need another check here: verify if write.format.default is ORC. If this is changed from within Hive or anything using our HMS client then of course ORC_FILES_ONLY is maintained, but if changed from just Iceberg API it will not reflect the changes.

Reply:
Done